connector_ecommerce/components/sale_order_onchange.py

```diff
@@ -44,6 +45,9 @@ def play_onchanges(self, model, values, onchange_fields):
             )
             all_values.update(new_values)
 
+        for fld in not_values:
+            if fld not in all_values:
```

Comment:
Will this condition ever be false? Before, you define both dictionaries based on the list of fields for a model, so AFAICS an element in one of them will never exist on the other, right?

That would make the check unnecessary, but I might be seeing it wrong… 😅

Reply:
I believe this has been clarified and is necessary due to `onchange_values`, right?

Reply:
this is because after applying onchange. [HERE](https://github.com/OCA/connector-ecommerce/pull/62/files/80317cee37d322329c88c06e721cb152ccdd32e8..e9be4e72d5a27320328b1b3112bd2d4603011705#diff-655a37a2a57355d263c191d9d14a7f764edcb6b5a90a11d1a71eea8ef006317bR46) the all_values has been changed. and updated accordingly.